### PR TITLE
Default ORDER BY for all entities and their relationships

### PR DESCRIPTION
**Pull request by coldic3**
| Q               | A                                                            |
|-----------------|--------------------------------------------------------------|
| Branch?         | 1.11 |
| Bug fix?        | yes (fixes sorting rows for postgres)                                                       |
| New feature?    | yes (adds default order by)                                                       |
| BC breaks?      | no                                                       |
| Deprecations?   | no |
| Related tickets |  |
| License         | MIT                                                          |

This PR introduces `OrderByIdentifierSqlWalker` and adds the default `order by` to the ORM configurations for all relationships.

### CODE REVIEW

**Review by GSadee — APPROVED** [review]
I am wondering whether this should be regarded as a bug fix or a new feature. In theory, this PR adds new functionality but also fixes an inconsistency in how data is loaded between different database engines, which fails some builds. I would probably be in favour of opening this to 1.11, but I'm curious to hear other people's opinions 🤔

**Comment by GSadee** [comment]
What is more, if the base branch were to be changed (or not only then), it would be nice to note this change in the proper UPGRADE files.

File: src/Sylius/Bundle/CoreBundle/SyliusCoreBundle.php [hunk 1]
```diff
@@ -61,6 +63,8 @@ public function boot(): void
     {
         parent::boot();
 
+        $this->setSqlWalker();
+
```
Comment (jakubtobiasz) [inline src/Sylius/Bundle/CoreBundle/SyliusCoreBundle.php]: Why not a compiler pass?

**Review by jakubtobiasz — APPROVED** [review]

File: src/Sylius/Bundle/CoreBundle/SyliusCoreBundle.php (hunk 1 above)
Comment (coldic3) [reply src/Sylius/Bundle/CoreBundle/SyliusCoreBundle.php]: It has to be done in the boot method. I tried with a compiler pass but it didn't work.

**Review by NoResponseMate — APPROVED** [review]

**Review by GSadee — APPROVED** [review]

**Comment by GSadee** [comment]
Thanks, Kevin! :1st_place_medal:

**Pull request merged**